File: VB/DocumentViewerWithToolbars/Reports/MyXtraReport.vb
Imports System
Imports DevExpress.XtraPrinting

Namespace DocumentViewerWithToolbars.Reports
    Partial Public Class MyXtraReport
        Inherits DevExpress.XtraReports.UI.XtraReport

        Private progress As Integer

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub MyXtraReport_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs) Handles MyBase.BeforePrint
            ReportPrintOptions.DetailCountOnEmptyDataSource = Math.Max(Math.Min(CInt((pageCount.Value)), 100), 1)
            AddHandler PrintingSystem.ProgressReflector.PositionChanged, AddressOf ProgressReflector_PositionChanged
        End Sub

        Private Sub MyXtraReport_AfterPrint(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.AfterPrint
            RemoveHandler PrintingSystem.ProgressReflector.PositionChanged, AddressOf ProgressReflector_PositionChanged
        End Sub

        Private Sub ProgressReflector_PositionChanged(ByVal sender As Object, ByVal e As EventArgs)
            Dim progressReflector = DirectCast(sender, ProgressReflector)
            progress = progressReflector.Position
        End Sub

        Private Sub xrShape1_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs) Handles xrShape1.BeforePrint
            xrShape1.Angle = progress * (180 \ 100)
        End Sub
    End Class
End Namespace

File: CS/DocumentViewerWithToolbars/Reports/MyXtraReport.cs
using System;
using DevExpress.XtraPrinting;

namespace DocumentViewerWithToolbars.Reports
{
    public partial class MyXtraReport : DevExpress.XtraReports.UI.XtraReport
    {
        int progress;

        public MyXtraReport()
        {
            InitializeComponent();
        }

        void MyXtraReport_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e)
        {
            ReportPrintOptions.DetailCountOnEmptyDataSource = Math.Max(Math.Min((int)pageCount.Value, 100), 1);
            PrintingSystem.ProgressReflector.PositionChanged += ProgressReflector_PositionChanged;
        }

        void MyXtraReport_AfterPrint(object sender, EventArgs e)
        {
            PrintingSystem.ProgressReflector.PositionChanged -= ProgressReflector_PositionChanged;
        }

        void ProgressReflector_PositionChanged(object sender, EventArgs e)
        {
            var progressReflector = (ProgressReflector)sender;
            progress = progressReflector.Position;
        }

        void xrShape1_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e)
        {
            xrShape1.Angle = progress * (180 / 100);
        }
    }
}
